File: project/core/module/lane/reconciliation/src/main/java/vdi/lane/reconciliation/SyncIndicator.java
package vdi.lane.reconciliation;

public class SyncIndicator {
  private boolean metaOutOfSync;
  private boolean sharesOutOfSync;
  private boolean installOutOfSync;

  public SyncIndicator() {
  }

  public SyncIndicator(boolean metaOutOfSync, boolean sharesOutOfSync, boolean installOutOfSync) {
    this.metaOutOfSync = metaOutOfSync;
    this.sharesOutOfSync = sharesOutOfSync;
    this.installOutOfSync = installOutOfSync;
  }

  public boolean isMetaOutOfSync() {
    return metaOutOfSync;
  }

  public void setMetaOutOfSync(boolean metaOutOfSync) {
    this.metaOutOfSync = metaOutOfSync;
  }

  public boolean isSharesOutOfSync() {
    return sharesOutOfSync;
  }

  public void setSharesOutOfSync(boolean sharesOutOfSync) {
    this.sharesOutOfSync = sharesOutOfSync;
  }

  public boolean isInstallOutOfSync() {
    return installOutOfSync;
  }

  public void setInstallOutOfSync(boolean installOutOfSync) {
    this.installOutOfSync = installOutOfSync;
  }

  public boolean isFullyOutOfSync() {
    return this.metaOutOfSync && this.sharesOutOfSync && this.installOutOfSync;
  }

  public boolean isOutOfSync() {
    return this.metaOutOfSync || this.sharesOutOfSync || this.installOutOfSync;
  }
}

File: project/core/module/lane/reconciliation/src/main/kotlin/vdi/lane/reconciliation/SyncIndicator.kt
package vdi.lane.reconciliation

internal data class SyncIndicator(
  /**
   * Indicates whether any installable meta files are out of sync with an
   * install target.
   *
   * Meta files include the VDI dataset metadata JSON, or any data dictionary
   * mapping files.
   */
  var metaOutOfSync: Boolean,

  /**
   * Indicates whether any share information files are out of sync with an
   * install target.
   */
  var sharesOutOfSync: Boolean,

  /**
   * Indicates whether a dataset's installable data is out of sync with an
   * install target.
   */
  var installOutOfSync: Boolean,
) {
  inline val fullyOutOfSync
    get() = metaOutOfSync && sharesOutOfSync && installOutOfSync

  inline val outOfSync
    get() = metaOutOfSync || sharesOutOfSync || installOutOfSync
}
